astropy/modeling/fitting.py

```diff
@@ -415,19 +415,23 @@ def __call__(self, model, x, y, z=None, weights=None, **kwargs):
 
         fitted_model = self.fitter(model, x, y, z, weights, **kwargs)
         if z is None:
+            fitted_model = self.fitter(fitted_model, x, y, **kwargs)
```

Comment:
Sorry if I'm mistaken, but I don't quite understand the reason of this line. Doesn't ``fitted_model = self.fitter(model, x, y, z, weights, **kwargs)`` before the ``if`` clause already suffice?